src/libutil/sysutil.cpp

```diff
@@ -376,18 +376,34 @@ bool enableVTMode()
     }
     return true;
 }
-#else
-bool enableVTMode() { return true; }
 #endif
 
 
+
 Term::Term (const std::ostream &stream)
 {
     m_is_console = (&stream == &std::cout && isatty(fileno(stdout)))
                 || (&stream == &std::cerr && isatty(fileno(stderr)))
                 || (&stream == &std::clog && isatty(fileno(stderr)));
-    if (is_console())
+
+#ifdef _WIN32
+    if (m_is_console())
         enableVTMode();
+#else
+    // Non-windows: also check the TERM env variable for a terminal known to
+    // be capable of the color codes. List copied from the Apache-licensed
+    // Google Benchmark: https://github.com/google/benchmark/blob/master/src/colorprint.cc
+    const char* const supported_terminals[] = {
+        "cygwin", "linux", "rxvt-unicode", "rxvt-unicode-256color",
```

Comment:
Perhaps asking tput on Linux would be better than using a hard coded list?  It has all the plumbing to dig through the system's termcap database stuff

will@Tower:~$ tput colors
8
will@Tower:~$ TERM=vt100 tput colors
-1
will@Tower:~$ TERM=xterm-256color tput colors
256

Reply:
That's not a bad idea, I guess I can try that. 

Realistically, is anybody using a terminal that doesn't support the basic color text usage that our Term class needs?

The current (pre-this-PR) code doesn't check at all, it just assumes all terminals will work. Nobody has complained, though to be fair I think the only place we use it is in tests, not in any critical path. 

I just happened to be browsing through the code for Google test and benchmark projects (for other reasons) and ran across this and borrowed it. I figured it was a step up from a blanket assumption, and there was no harm, and if it's good enough for Google, it probably is good enough for us.

So the question is just whether it's worth the trouble, and whether there are any cases where tput will not reliably be found? (I guess if not able to run tput, it can fall back on this test.)